pandas/core/internals.py

```diff
@@ -1845,12 +1845,13 @@ def is_bool(self):
         """
         return lib.is_bool_array(self.values.ravel())
 
-    # TODO: Refactor when convert_objects is removed since there will be 1 path
     def convert(self, *args, **kwargs):
-        """ attempt to coerce any object types to better types return a copy of
-        the block (if copy = True) by definition we ARE an ObjectBlock!!!!!
+        """
+        Attempt to coerce any object types to more specific data types.
```

Comment:
there's a whole bunch of code here that i think u can ditch now that was for compat with convert_objects